js_modules/dagster-ui/packages/ui-core/src/app/AppProvider.tsx

```diff
@@ -110,13 +112,17 @@ export const AppProvider = (props: AppProviderProps) => {
     return new RetryLink({
       attempts: {
         max: 2,
-        retryIf: (error, _operation) => {
-          return error && error.statusCode && [502, 503, 504].includes(error.statusCode);
+        retryIf: async (error, _operation) => {
+          if (error && error.statusCode && httpStatusCodesToRetry.includes(error.statusCode)) {
+            const wait = parseInt(error?.response?.headers?.get?.('retry-after') ?? '300');
```

Comment:
300 seconds seems like an aggressive default if nothing explicit is set 

nit: pull out and name as const

Reply:
ah setTimeout is in ms,  the header is in seconds though
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

so well need to convert